### PR TITLE
Allow passing custom ocr object to arynpartitioner

### PR DESCRIPTION
**Pull request by baitsguy**
Let's you pass in a configured object, the current implementation would allow you to configure the model

Tested with test

### CODE REVIEW

File: lib/sycamore/sycamore/transforms/detr_partitioner.py [hunk 1]
```diff
@@ -542,16 +545,19 @@ def _run_text_extractor_document(
         use_cache: bool,
         use_ocr: bool,
         ocr_images: bool,
-        text_extractor_model: str,
+        text_extractor_model: Union[str, OcrModel],
```
Comment (karanataryn) [inline lib/sycamore/sycamore/transforms/detr_partitioner.py:548]: Can we make this an instance of `TextExtractor`? I could imagine wanting to configure PDFMiner similarly as well.

File: lib/sycamore/sycamore/tests/unit/transforms/test_detr_partitioner.py [hunk 2]
```diff
@@ -61,6 +64,17 @@ def test_partition(self):
         d = check_partition(s, TEST_DIR / "resources/data/pdfs/basic_table.pdf", use_ocr=True, use_cache=False)
         assert len(d) == 1
 
+    def test_partition_w_ocr_instance(self):
```
Comment (karanataryn) [inline lib/sycamore/sycamore/tests/unit/transforms/test_detr_partitioner.py:67]: nit: `test_partition_with_ocr_instance`

**Review by karanataryn — APPROVED** [review]
LGTM. Have a few suggestions

File: lib/sycamore/sycamore/transforms/detr_partitioner.py (hunk 1 above)
Comment (baitsguy) [reply lib/sycamore/sycamore/transforms/detr_partitioner.py:548]: Sorry for the quick merge. Let's update this as the need arises, it shouldn't cause any issues

**Pull request merged**